crowdstrikeoauthapi_connector.py

```diff
@@ -1556,6 +1556,97 @@ def _handle_remove_hosts(self, param):
 
         return action_result.set_status(phantom.APP_SUCCESS, "Host removed successfully")
 
+    def _handle_create_ioa_rule(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        try:
+            field_values = json.loads(param['field_values'])
+        except json.JSONDecodeError as e:
+            return action_result.set_status(
+                phantom.APP_ERROR,
+                f'Failed to parse field_values: {e}'
+            )
+
+        create_comment = param.get('comment')
+
+        create_params = {
+            'rulegroup_id': param['rule_group_id'],
+            'name': param['name'],
+            'description': param['description'],
+            'pattern_severity': param['severity'],
+            'ruletype_id': str(param['rule_type_id']),
+            'disposition_id': param['disposition_id'],
+            'field_values': field_values
+        }
+        if create_comment:
+            create_params['comment'] = create_comment
+
+        ret_val, resp_json = self._make_rest_call_helper_oauth2(
+            action_result,
+            CROWDSTRIKE_IOA_CREATE_RULE_ENDPOINT,
+            json_data=create_params,
+            method='post'
+        )
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        rulegroup_id = resp_json['resources'][0].get('rulegroup_id')
```

Comment:
Is there ever a chance that the 'resources' key will not exist in the response JSON? I'm just asking because in the _make_rest_call_oauth2 I only see a an error returned if we can't connect to the server. I have seen in the past where the the response JSON is empty.

Reply:
I don't know of any case where we'd get a "success" HTTP code (checked above), but get a response with the resources key missing. That would probably indicate a major incident with CrowdStrike's service.